File: DSA/SakshamDwivedi_590013147/Day18/D18P3.c
#include <stdio.h>

int colours[100];
int size;
int high, low, mid;

int main(){
    printf("Enter the size of the colours array (Less than 100) : ");
    scanf("%d", &size);
    high = size - 1;
    printf("Enter the colours [0, 1 and 2] : ");
    for(int i = 0; i < size; i++){
        scanf("%d", &colours[i]);
    }
    while(mid <= high){
        if(colours[mid] == 0){
            int temp = colours[mid];
            colours[mid] = colours[low];
            colours[low] = temp;
            mid++;
            low++;
        }
        else if(colours[mid] == 1){
            mid++;
        }
        else{
            int temp = colours[mid];
            colours[mid] = colours[high];
            colours[high] = temp;
            high--;
        }
    }

    printf("Sorted Array : ");
    for(int i = 0; i < size; i++){
        printf("%d ", colours[i]);
    }
}

File: DSA/SakshamDwivedi_590013147/Day18/D18P3.py
def dutch_flag_sort(colours):
    low = 0
    mid = 0
    high = len(colours) - 1

    while(mid <= high):
        if(colours[mid] == 0):
            colours[mid], colours[low] = colours[low], colours[mid]
            mid += 1
            low += 1
        elif(colours[mid] == 1):
            mid+=1
        else:
            colours[mid], colours[high] = colours[high], colours[mid]
            high -= 1
    return colours

#Fixed Input
print(dutch_flag_sort([2,0,2,1,1,0]))

#Custom Input
arr = list(int(x) for x in input("Enter the colours [0,1 and 2 only] : ").split())
print(dutch_flag_sort(arr))
